partiql-eval/src/test/kotlin/org/partiql/eval/internal/PartiQLEngineDefaultTest.kt

```diff
@@ -155,6 +160,123 @@ class PartiQLEngineDefaultTest {
+                """.trimIndent(),
+                expected = bagValue<PartiQLValue>()
+            ),
+            // LEFT OUTER JOIN -- No Matches
+            SuccessTestCase(
+                input = """
+                    SELECT VALUE [lhs, rhs]
+                    FROM << 0, 1, 2 >> lhs
+                    LEFT OUTER JOIN << 3, 4, 5 >> rhs
+                    ON lhs = rhs
+                """.trimIndent(),
+                expected = bagValue(
+                    listValue(int32Value(0), int32Value(null)),
+                    listValue(int32Value(1), int32Value(null)),
+                    listValue(int32Value(2), int32Value(null)),
+                )
+            ),
+            // RIGHT OUTER JOIN -- Easy
+            SuccessTestCase(
+                input = """
+                    SELECT VALUE [lhs, rhs]
+                    FROM << 0, 1, 2 >> lhs
+                    RIGHT OUTER JOIN << 0, 2, 3 >> rhs
+                    ON lhs = rhs
+                """.trimIndent(),
+                expected = bagValue(
+                    listValue(int32Value(0), int32Value(0)),
+                    listValue(int32Value(2), int32Value(2)),
+                    listValue(int32Value(null), int32Value(3)),
+                )
+            ),
+            // RIGHT OUTER JOIN -- RHS Empty
+            SuccessTestCase(
+                input = """
+                    SELECT VALUE [lhs, rhs]
+                    FROM << 0, 1, 2 >> lhs
+                    RIGHT OUTER JOIN <<>> rhs
+                    ON lhs = rhs
+                """.trimIndent(),
+                expected = bagValue<PartiQLValue>()
+            ),
+            // RIGHT OUTER JOIN -- LHS Empty
+            SuccessTestCase(
+                input = """
+                    SELECT VALUE [lhs, rhs]
+                    FROM (
+                        SELECT VALUE n
+                        FROM << 0, 1, 2 >> AS n
+                        WHERE n > 100
+                    ) lhs RIGHT OUTER JOIN
+                        << 0, 2, 3>> rhs
+                    ON lhs = rhs
+                """.trimIndent(),
+                expected = bagValue<PartiQLValue>(
+                    listValue(int32Value(null), int32Value(0)),
+                    listValue(int32Value(null), int32Value(2)),
+                    listValue(int32Value(null), int32Value(3)),
+                )
+            ),
+            // RIGHT OUTER JOIN -- No Matches
+            SuccessTestCase(
+                input = """
+                    SELECT VALUE [lhs, rhs]
+                    FROM << 0, 1, 2 >> lhs
+                    RIGHT OUTER JOIN << 3, 4, 5 >> rhs
+                    ON lhs = rhs
+                """.trimIndent(),
+                expected = bagValue(
+                    listValue(int32Value(null), int32Value(3)),
+                    listValue(int32Value(null), int32Value(4)),
+                    listValue(int32Value(null), int32Value(5)),
+                )
+            ),
```

Comment:
Perhaps add a couple tests for full outer join as well? 

Also, would be nice if we can test the lateral properties of Left Join, and set up baseline for non-lateral behavior for Right and Full Join. Not sure if those should go to the evaluation test or the planning test (if not already exist)

Reply:
I've been using the conformance tests for full outer join -- but I just updated and pushed additional tests for planning.